src/components/Main/Main.tsx

```diff
@@ -14,19 +17,23 @@ const TitleTexts: string[] = [
 ];
 
 const getTitleText = (): string => {
-  //TODO: write real logic
+  //TODO: write real logic, when filtring options will be implemented
   return TitleTexts[0];
 };
 
-function Main({handlePostClick} : { handlePostClick: (post: PostData) => void }) {
+function Main({
+  handlePostClick,
+}: {
+  handlePostClick: (post: PostData) => void;
+}) {
+  const dispatch = useAppDispatch();
+  getPosts(localStorage.jwt).then((data) => dispatch(setInitialPosts(data)));
```

Comment:
You shouldn't call any function just inside the **component's body** because these functions are called every time the component rerenders causing spamming to the server. You need to place them inside `useEffect`